src/inapp/inapp.ts

```diff
@@ -523,7 +516,9 @@ export function getInAppMessages(
       showInAppMessagesAutomatically.display === 'deferred'
         ? {
             triggerDisplayMessages: (messages: Partial<InAppMessage>[]) => {
-              parsedMessages = messages as InAppMessage[];
+              parsedMessages = filterHiddenInAppMessages(
```

Comment:
Regardless of whether `getInAppMessages` call includes `showInAppMessagesAutomatically` param or not, we should still NOT paint messages that would be normally hidden (already read messages, silently delivered messages, and raw json messages)